platform/kobo/enable-wifi.sh

```diff
@@ -14,43 +14,84 @@ for fd in /proc/"$$"/fd/*; do
 done
 
 # Some platforms do *NOT* use sdio_wifi_pwr, even when it is physically there...
-SKIP_SDIO_PWR_MODULE=""
+POWER_TOGGLE="module"
 # We also want to choose the wpa_supplicant driver depending on the module...
 WPA_SUPPLICANT_DRIVER="wext"
+# And some platforms also put the modules in some funkier paths...
+KMOD_PATH="/drivers/${PLATFORM}/wifi"
 case "${WIFI_MODULE}" in
     "moal")
-        SKIP_SDIO_PWR_MODULE="1"
+        POWER_TOGGLE="ntx_io"
         WPA_SUPPLICANT_DRIVER="nl80211"
         ;;
+    "wlan_drv_gen4m")
+        POWER_TOGGLE="wmt"
+        WPA_SUPPLICANT_DRIVER="nl80211"
+        KMOD_PATH="/drivers/${PLATFORM}/mt66xx"
+        ;;
 esac
 
+# Load the requested module if it isn't already
+insmod_asneeded() {
+    kmod="${1}"
+    shift
+
+    if ! grep -q "^${kmod} " "/proc/modules"; then
+        insmod "${KMOD_PATH}/${kmod}.ko" "${@}"
+        usleep 250000
+    fi
+}
+
 # Power up WiFi chip
-if [ -n "${SKIP_SDIO_PWR_MODULE}" ]; then
-    # 208 is CM_WIFI_CTRL
-    ./luajit frontend/device/kobo/ntx_io.lua 208 1
-else
-    if ! grep -q "^sdio_wifi_pwr" "/proc/modules"; then
-        if [ -e "/drivers/${PLATFORM}/wifi/sdio_wifi_pwr.ko" ]; then
-            # Handle the shitty DVFS switcheroo...
-            if [ -n "${CPUFREQ_DVFS}" ]; then
-                echo "userspace" >"/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor"
-                echo "1" >"/sys/devices/platform/mxc_dvfs_core.0/enable"
-            fi
+case "${POWER_TOGGLE}" in
+    "ntx_io")
+        # 208 is CM_WIFI_CTRL
+        ./luajit frontend/device/kobo/ntx_io.lua 208 1
+        ;;
+    "wmt")
+        # NOTE: Unlike earlier platforms, it seems the WiFi modules are only loaded once, and never unloaded.
+        #       So, just make sure they actually are loaded before we go on...
+        insmod_asneeded "wmt_drv"
+        insmod_asneeded "wmt_chrdev_wifi"
+        insmod_asneeded "wmt_cdev_bt"
+        insmod_asneeded "${WIFI_MODULE}"
 
-            insmod "/drivers/${PLATFORM}/wifi/sdio_wifi_pwr.ko"
-        else
-            # Poke the kernel via ioctl on platforms without the dedicated power module...
-            # 208 is CM_WIFI_CTRL
-            ./luajit frontend/device/kobo/ntx_io.lua 208 1
+        # Black magic courtesy of wmt_dbg_func_ctrl @ (out of tree) modules/connectivity/wmt_mt66xx/common_main/linux/wmt_dbg.c
+        # Enable debug commands
+        echo "0xDB9DB9" >/proc/driver/wmt_dbg
+        # Disable the LPBK test
+        echo "7 9 0" >/proc/driver/wmt_dbg
+        # Nickel appears to sleep for ~1s
+        sleep 1
+        echo "0xDB9DB9" >/proc/driver/wmt_dbg
+        # Enable the LPBK test (this'll block for ~1.3s)
+        echo "7 9 1" >/proc/driver/wmt_dbg
+        # Finally, power on the chip
+        echo 1 >/dev/wmtWifi
```

Comment:
Heh wtf

Reply:
Yeah, don't ask me what LPBK stands for ;D.